syntax = "proto3";

package analytics;

service AnalyticsService {
	rpc RecordHit(HitRequest) returns (HitResponse);
    rpc GetStats(StatsRequest) returns (StatsResponse);
}

message HitRequest {
    string short_url = 1;
    string long_url = 2;
    int64 timestamp = 3;
    string hash = 4;
}

message HitResponse {
    bool success = 1;
    string message = 2;
}

message StatsRequest {
    string hash = 1;
}

message StatsResponse {
    repeated URLStats stats = 1;
}

message URLStats {
    string hash = 1;
    string short_url = 2;
    string long_url = 3;
    int32 hit_count = 4;
    int64 last_accessed = 5;
    int64 created_at = 6;
}